.circleci/config.yml

```diff
@@ -5,18 +5,42 @@ aliases:
 
 executors:
   sbt-executor:
-    working_directory: *working_directory
     docker:
       - image: openlaw/scala-builder
+    working_directory: *working_directory
 
 jobs:
   compile:
     executor: sbt-executor
     steps:
       - checkout
+       
+      #git and sbt caching
+      - restore_cache:
+          keys:
+            - source-v1-{{ .Branch }}-{{ .Revision }}
+            - source-v1-{{ .Branch }}-
+            - source-v1-
+            - sbt-cache
+          
       - run:
           name: Compile smoke tests
           command: sbt -Dfatal-warnings=true -Dscalac-optimize=true clean exit
+       
+      #git garbage collection 
+      - run: git gc --quiet
+
+      #git and sbt caching
```

Comment:
I don't think we need to save the git, because it's automatically handled by circleci.

Reply:
my bad, did not know about it
https://circleci.com/docs/2.0/caching/#source-caching